tests/providers/snowflake/transfers/test_s3_to_snowflake.py

```diff
@@ -26,6 +26,42 @@
 class TestS3ToSnowflakeTransfer(unittest.TestCase):
     @mock.patch("airflow.providers.snowflake.hooks.snowflake.SnowflakeHook.run")
     def test_execute(self, mock_run):
+        table = 'table'
+        stage = 'stage'
+        file_format = 'file_format'
+        schema = 'schema'
+
+        S3ToSnowflakeOperator(
+            s3_keys=None,
+            table=table,
+            stage=stage,
+            file_format=file_format,
+            schema=schema,
+            columns_array=None,
+            task_id="task_id",
+            dag=None,
+        ).execute(None)
+
+        base_sql = """
+                FROM @{stage}/
+
+                file_format={file_format}
+            """.format(
+            stage=stage, file_format=file_format
+        )
+
+        copy_query = """
+                COPY INTO {schema}.{table} {base_sql}
+            """.format(
+            schema=schema, table=table, base_sql=base_sql
+        )
+        copy_query = "\n".join(line.strip() for line in copy_query.splitlines())
+
+        mock_run.assert_called_once()
+        assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], copy_query)
+
+    @mock.patch("airflow.providers.snowflake.hooks.snowflake.SnowflakeHook.run")
+    def test_execute_with_s3_keys(self, mock_run):
```

Comment:
```suggestion
class TestS3ToSnowflakeTransfer(unittest.TestCase):
    @parameterized.expand([
        ('base', dict(table='table',
                      stage='stage',
                      file_format='file_format',
                      schema='schema',
                      ), "COPY INTO schema.table FROM @stage/ file_format=file_format"),
        ('files', dict(
            s3_keys=['1.csv', '2.csv'],
            table='table',
            stage='stage',
            file_format='file_format',
            schema='schema',
        ), "COPY INTO schema.table FROM @stage/ files=('1.csv', '2.csv') file_format=file_format"),
        ('columns', dict(
            table='table',
            stage='stage',
            file_format='file_format',
            schema='schema',
            columns_array=['col1', 'col2', 'col3'],
        ), "COPY INTO schema.table(col1,col2,col3) FROM @stage/ file_format=file_format"),
    ])
    @mock.patch("airflow.providers.snowflake.hooks.snowflake.SnowflakeHook.run")
    def test_execute(self, name, kwargs, expected, mock_run):
        S3ToSnowflakeOperator(
            **kwargs,
            task_id="task_id",
            dag=None,
        ).execute(None)

        mock_run.assert_called_once()
        assert_equal_ignore_multiple_spaces(self, mock_run.call_args[0][0], expected)
```
entire test can be replaced with this
using parameterization makes it much easier to read IMO

Reply:
Agreed, update the test case to be parametrized.